core/tauri/src/api/process.rs

```diff
@@ -82,10 +82,49 @@ pub fn restart(env: &Env) {
   use std::process::{exit, Command};
 
   if let Ok(path) = current_binary(env) {
-    Command::new(path)
-      .args(&env.args)
-      .spawn()
-      .expect("application failed to start");
+    // on macOS on updates the binary name might have changed
+    // so we'll read the Contents/MacOS folder instead to infer the actual binary path
```

Comment:
Can we read `Contents/Info.plist` to determine new binary name?

Reply:
ohhhhhh we can.. let's do that

Reply:
Amr the no-Apple dev that knows more about Apple than any other dev